File: app/src/main/java/com/hippo/ehviewer/dao/BookmarksDao.java
package com.hippo.ehviewer.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface BookmarksDao extends BasicDao<BookmarkInfo> {
    @Insert
    long insert(BookmarkInfo bookmark);

    @Delete
    void delete(BookmarkInfo bookmark);

    @Query("SELECT * FROM BOOKMARKS ORDER BY TIME DESC")
    List<BookmarkInfo> list();
}

File: app/src/main/java/com/hippo/ehviewer/dao/BookmarksDao.kt
package com.hippo.ehviewer.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface BookmarksDao : BasicDao<BookmarkInfo> {
    @Insert
    override fun insert(t: BookmarkInfo): Long

    @Delete
    fun delete(bookmark: BookmarkInfo)

    @Query("SELECT * FROM BOOKMARKS ORDER BY TIME DESC")
    override fun list(): List<BookmarkInfo>
}